app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala

```diff
@@ -16,12 +16,90 @@
 
 package uk.gov.hmrc.nationalinsurancerecord.connectors
 
-import uk.gov.hmrc.nationalinsurancerecord.domain.nps.{NpsLiability, NpsNIRecord, NpsSummary}
+import play.api.data.validation.ValidationError
+import play.api.libs.json.{JsPath, Reads}
+import uk.gov.hmrc.domain.Nino
+import uk.gov.hmrc.nationalinsurancerecord.domain.nps.{NpsLiabilities, NpsNIRecord, NpsSummary}
+import uk.gov.hmrc.play.config.ServicesConfig
+import uk.gov.hmrc.play.http.{HeaderCarrier, HttpGet, HttpReads, HttpResponse}
+import uk.gov.hmrc.nationalinsurancerecord.WSHttp
+import uk.gov.hmrc.nationalinsurancerecord.domain.APITypes
+import uk.gov.hmrc.nationalinsurancerecord.domain.APITypes.APITypes
+import uk.gov.hmrc.nationalinsurancerecord.util.NIRecordConstants
 
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
+object NpsConnector extends NpsConnector with ServicesConfig {
+  override val serviceUrl = baseUrl("nps-hod")
+  override val serviceOriginatorIdKey = getConfString("nps-hod.originatoridkey", "")
+  override val serviceOriginatorId = getConfString("nps-hod.originatoridvalue", "")
+  override def http: HttpGet = WSHttp
+}
 
 trait NpsConnector {
-  def getNationalInsuranceRecord: Future[NpsNIRecord]
-  def getLiabilities: Future[List[NpsLiability]]
-  def getSummary: Future[NpsSummary]
+
+  val serviceUrl: String
+  val serviceOriginatorIdKey: String
+  val serviceOriginatorId: String
+
+  class JsonValidationException(message: String) extends Exception(message)
+
+  def http: HttpGet
+  def url(path: String): String = s"$serviceUrl$path"
+  def requestHeaderCarrier(implicit hc: HeaderCarrier): HeaderCarrier = hc.withExtraHeaders(serviceOriginatorIdKey -> serviceOriginatorId)
+  private def ninoWithoutSuffix(nino: Nino): String = nino.value.substring(0, NIRecordConstants.ninoLengthWithoutSuffix)
+
+  def getLiabilities(nino: Nino)(implicit hc: HeaderCarrier): Future[NpsLiabilities] = {
```

Comment:
Personally I would change this method to return the List[NpsLiabilities] and map the response in this method.  But that's not required.